process/block/sovereignChainBlock_test.go

```diff
@@ -343,4 +400,283 @@ func TestSovereignChainBlockProcessor_createAndSetOutGoingMiniBlock(t *testing.T
+				return expectedErr
+			},
+		}
+		sovArgs := createArgsSovereignChainBlockProcessor(arguments)
+		processHandler := arguments.CoreComponents.ProcessStatusHandler()
+		mockProcessHandler := processHandler.(*testscommon.ProcessStatusHandlerStub)
+		busyIdleCalled := make([]string, 0)
+		mockProcessHandler.SetIdleCalled = func() {
+			busyIdleCalled = append(busyIdleCalled, idleIdentifier)
+		}
+		mockProcessHandler.SetBusyCalled = func(reason string) {
+			busyIdleCalled = append(busyIdleCalled, busyIdentifier)
+		}
+		expectedBusyIdleSequencePerCall := []string{busyIdentifier, idleIdentifier}
+
+		scbp, err := blproc.NewSovereignChainBlockProcessor(sovArgs)
+		require.Nil(t, err)
+
+		doesHaveTime := func() bool {
+			return true
+		}
+
+		expectedSovHeader := &block.SovereignChainHeader{
+			Header: &block.Header{
+				Nonce: 37,
+				Round: 38,
+				Epoch: 1,
+			},
+		}
+
+		hdr, bodyHandler, err := scbp.CreateBlock(expectedSovHeader, doesHaveTime)
+		require.True(t, check.IfNil(bodyHandler))
+		require.True(t, check.IfNil(hdr))
+		require.Equal(t, expectedErr, err)
+		require.Equal(t, expectedBusyIdleSequencePerCall, busyIdleCalled)
+	})
+	t.Run("should work with sovereign header and epoch start rewriting the epoch value", func(t *testing.T) {
+		currentEpoch := uint32(1)
+		nextEpoch := uint32(2)
+
+		coreComponents, dataComponents, bootstrapComponents, statusComponents := createComponentHolderMocks()
+		arguments := createSovereignMockArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+		arguments.EpochStartTrigger = &testscommon.EpochStartTriggerStub{
+			IsEpochStartCalled: func() bool {
+				return true
+			},
+			MetaEpochCalled: func() uint32 {
+				return nextEpoch
+			},
+		}
+		sovArgs := createArgsSovereignChainBlockProcessor(arguments)
+		processHandler := arguments.CoreComponents.ProcessStatusHandler()
+		mockProcessHandler := processHandler.(*testscommon.ProcessStatusHandlerStub)
+		busyIdleCalled := make([]string, 0)
+		mockProcessHandler.SetIdleCalled = func() {
+			busyIdleCalled = append(busyIdleCalled, idleIdentifier)
+		}
+		mockProcessHandler.SetBusyCalled = func(reason string) {
+			busyIdleCalled = append(busyIdleCalled, busyIdentifier)
+		}
+		expectedBusyIdleSequencePerCall := []string{busyIdentifier, idleIdentifier}
+
+		scbp, err := blproc.NewSovereignChainBlockProcessor(sovArgs)
+		require.Nil(t, err)
+
+		sovHeader := &block.SovereignChainHeader{
+			Header: &block.Header{
+				Nonce: 37,
+				Round: 38,
+				Epoch: currentEpoch,
+			},
+		}
+		expectedSovHeader := &block.SovereignChainHeader{
+			Header: &block.Header{
+				Nonce: 37,
+				Round: 38,
+				Epoch: nextEpoch,
+			},
+			IsStartOfEpoch: true,
+		}
+		doesHaveTime := func() bool {
+			return true
+		}
+
+		hdr, bodyHandler, err := scbp.CreateBlock(sovHeader, doesHaveTime)
+		require.False(t, check.IfNil(bodyHandler))
```

Comment:
Check directly with miniblocks and skip using these False.Nil.
Applies also for expected header. If you require it to be equal to your expected object, there is no need to also check it is not nil